modules/account-lib/src/coin/xrp/transactionBuilder.ts

```diff
@@ -1,57 +1,231 @@
 import BigNumber from 'bignumber.js';
+import RippleBinaryCodec from 'ripple-binary-codec';
+import { ApiMemo } from 'ripple-lib/dist/npm/transaction/utils';
+import * as rippleTypes from 'ripple-lib/dist/npm/transaction/types';
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseTransactionBuilder } from '../baseCoin';
-import { NotImplementedError } from '../baseCoin/errors';
-import { BaseAddress, BaseKey } from '../baseCoin/iface';
+import { BaseTransactionBuilder, TransactionType } from '../baseCoin';
+import { BuildTransactionError, InvalidTransactionError } from '../baseCoin/errors';
+import { BaseAddress, BaseFee, BaseKey } from '../baseCoin/iface';
 import { Transaction } from './transaction';
-
+import { AddressValidationError } from './errors';
+import { KeyPair } from './keyPair';
+import utils from './utils';
+import { BaseTransactionSchema } from './txnSchema';
 export abstract class TransactionBuilder extends BaseTransactionBuilder {
-  private _transaction: Transaction;
+  protected _transaction: Transaction;
+  protected _keyPairs: KeyPair[];
+
+  protected _sender: string;
+  protected _memos?: { Memo: ApiMemo }[];
+  protected _flags?: number;
+  protected _fulfillment?: string;
+  protected _lastLedgerSequence?: number;
+  protected _fee?: string;
+  protected _sequence?: number;
 
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
+    this._transaction = new Transaction(_coinConfig);
+    this._keyPairs = [];
+  }
+
+  /**
+   * Sets the fee.
+   *
+   *
+   * @param {BaseFee} feeObj The amount to pay to the fee sink
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   */
+  fee(feeObj: BaseFee): this {
+    this.validateValue(new BigNumber(feeObj.fee));
+    this._fee = feeObj.fee;
+    return this;
   }
 
   /**
-   * Initialize the transaction builder fields using the decoded transaction data
+   * Sets the transaction sender.
+   *
+   * @param {BaseAddress} sender The sender account
+   * @returns {TransactionBuilder} This transaction builder.
    *
-   * @param {Transaction} tx the transaction data
    */
-  initBuilder(tx: Transaction): void {
-    throw new NotImplementedError('initBuilder not implemented');
+  sender(sender: BaseAddress): this {
+    this.validateAddress(sender);
+    this._sender = sender.address;
+    this._transaction.sender(sender.address);
```

Comment:
Why do you need to store this in 2 places?

Reply:
Removed from this._transaction